homeassistant/components/sensoterra/sensor.py

```diff
@@ -0,0 +1,135 @@
+"""Sensoterra devices."""
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    EntityCategory,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity, UpdateFailed
+
+from .const import CONFIGURATION_URL, DOMAIN
+from .coordinator import SensoterraCoordinator, SensoterraSensor
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_devices: AddEntitiesCallback
+) -> None:
+    """Set up Sensoterra sensor."""
+
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    def _add_devices(devices: dict[str, SensoterraSensor]) -> None:
+        sensors = [
+            SensoterraEntity(coordinator, sensor_id, sensor)
+            for sensor_id, sensor in devices.items()
+            if sensor.type
+            in ["MOISTURE", "SI", "TEMPERATURE", "BATTERY", "RSSI", "LASTSEEN"]
+        ]
+        async_add_devices(sensors)
+
+    coordinator.add_devices_callback = _add_devices
+
+    _add_devices(coordinator.data)
+
+
+class SensoterraEntity(CoordinatorEntity, SensorEntity):
```

Comment:
Please type the Coordinator entity

```suggestion
class SensoterraSensor(CoordinatorEntity[..], SensorEntity):
```